source/common/router/config_impl.h

```diff
@@ -404,8 +408,10 @@ class RetryPolicyImpl : public RetryPolicy {
   std::vector<uint32_t> retriable_status_codes_;
   std::vector<Http::HeaderMatcherSharedPtr> retriable_headers_;
   std::vector<Http::HeaderMatcherSharedPtr> retriable_request_headers_;
-  absl::optional<std::chrono::milliseconds> base_interval_;
-  absl::optional<std::chrono::milliseconds> max_interval_;
+  // Durations below are optional in the config. -1 sentinel is used to represent a value that
+  // was not provided.
+  std::chrono::milliseconds base_interval_{-1};
```

Comment:
suggest
```
static constexpr std::chrono::milliseconds TimeoutUnset = -1;
```
or if std:;chrono::miliseconds cannot be constexpr-constructed then you could use an inline function to supply the Unset value to use as a default.

Reply:
Done

Reply:
I didn't check, but it would be good to ensure that we have PGV rules that prevent the user from ever setting a sentinal value in the config.

Reply:
I didn't see PGV rules around `Durations`, I assume it's implied that a negative duration is unreasonable.